### DISCUSSION TITLE
Autoscale in different namespaces

### DISCUSSION THREAD

**Discussion by cambyn** [kedacore/http-add-on#566] [Q&A]
Can we use http-add-on to scale pods to 0 in different namespaces? If the answer is yes, then what will be a suitable way to so. Thanks

**Comment by tomkerkhove** [comment]
As far as I know, yes. However, all traffic needs to be intercepted by KEDA HTTP add-on.

**Comment by HongboFei** [comment]
I would say absolutely yes, that's why we use http-add on.

**Reply by tomkerkhove** [reply]
@HongboFei Any way we can improve our docs to help people such as @cambyn to educate/raise this better?

**Reply by ishallbethat** [reply]
@HongboFei @tomkerkhove  Could you please share how to accomplish this autoscaling in different namespaces ?

**Reply by brondum** [reply]
@ishallbethat Did you ever find a way of doing the cross namespace stuff?

**Reply by MohamadTahir** [reply]
Any progress with this? I have keda and keda HTTP add-on in the `keda` namespace and my deployments and services are in the `test` namespace, I have nginx ingress controller which is in the same namespace as my deployment and service.
I can not route the request from my ingress to `keda-add-ons-http-interceptor-proxy` as it is in keda namespace (different namespace).

**Reply by danielamar101** [reply]
I second @MohamadTahir's problem

**Reply by danielamar101** [reply]
Actually, @MohamadTahir , I discovered a solution to this problem. I have the ingress resource live in the same namespace as wherever the http addon is installed to, while keeping the deployments I want to scale in any namespace.. This does the trick, as the HttpScaledObject  tells the interceptor what deployment/svc to redirect the request to.

**Reply by tomkerkhove** [reply]
Sorry for the slow response, I lost track of notifications.

So you are both having issues if the ingress is not in the same namespace as the operator's installation?

**Reply by danielamar101** [reply]
Essentially yes, as the operator installs the proxy service in the same namespace, which our deployment ingresses need, if that makes sense.

**Reply by ishallbethat** [reply]
@brondum i didn't solve it.
@danielamar101 Can you pls be more specific how I can use scaleobject to achieve this

**Reply by danielamar101** [reply]
So the http-add-on helm chart installs three deployments: 
1. The controller-manager
2. external scaler
3. interceptor

And a few services:
1. http-controller-manager-metrics-service
2. external scaler
3. interceptor-admin
4. **interceptor-proxy**
5. operator-admin

In order to route external traffic to the deployment, I use an ingress and specify my own service that will route to the deployment, however for the add-on to work, we have to route the ingress to the **interceptor-proxy** service. 

Typically this interceptor-proxy service lives in a namespace like `keda-http` while my ingress would be in another namespace. **It is currently impossible to have an ingress route to a service in another namespace without super hacky stuff.** So instead I put my ingress in the same namespace as the `interceptor-proxy` service (in my case `keda-http`). The HttpScaledObject then tells the interceptor which deployment to route the traffic to, **as this is no longer implied** by having an ingress route to my own service which is selected by my app deployment.

**Reply by MohamadTahir** [reply]
This is not a clean solution, I prefer keeping all the resources that are related to a named project in their own namespace, there is no need for me to put the ingress in the add-on namespace while keeping the rest of the components in a different namespace. this can get messy.

**Reply by HongboFei** [reply]
@tomkerkhove, let me see if i can find a better example. But i agree with @ishallbethat  and @MohamadTahir  we have to put the ingress object in the namespace same with the http-addon installed for now.

**Reply by danielamar101** [reply]
@MohamadTahir I agree with you, but in the current architecture I don’t see any another way. 

@HongboFei are you referring to the ingress for a deployment? Because thats what I’m implying in my comment above, when saying the deployment ingresses need to be in the same namespace as the addon, due to k8s limitations.

**Comment by sudermanjr** [comment]
I wonder if it would be possible to deploy multiple interceptor proxies in multiple namespaces? This would allow the ingress for each service to live in the namespace for the service, as well as spread the interceptor load across multiple interceptors in large deployments.

**Reply by cldfzn** [reply]
Yeah, this is essential for multi-tenant environments where tenants are limited to isolated namespaces. Without this we'll need to create a CRD and operator to create ingresses in the desired namespace from their tenant namespace.
